File: ruby/problem6.rb

def sum_of_squares
  (1..100).inject{ |sum,n| sum + n**2}
end

def sum
  (1..100).inject{ |sum,n| sum + n}
end
def
 square_of_sums
  sum ** 2
end

def difference
  square_of_sums - sum_of_squares
end

File: problem6.py
def square_of_sum(end_num, result = 0):
  if (end_num == 0):
    return result ** 2
  else:
    result += end_num
    return square_of_sum(end_num - 1, result)

def sum_of_squares(end_num,result = 0):
  if (end_num == 0):
    return result
  else:
    result += end_num ** 2
    return sum_of_squares(end_num - 1, result)

def difference(num):
  return square_of_sum(num) - sum_of_squares(num)

print difference(100)
